File: book-model/src/main/java/org/gdgcd/demo/service/BookMetaEnvelope.java
package org.gdgcd.demo.service;


import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;
import java.util.List;

public class BookMetaEnvelope implements Page<BookMeta> {
    @JsonProperty("Error")
    public String errorNo;

    @JsonProperty("Time")
    public Float responseTime;

    @JsonProperty("Total")
    public String totalCount;

    @JsonProperty("Page")
    public Integer pageNo;

    @JsonProperty("Books")
    public List<BookMeta> bookList = new ArrayList<BookMeta>();

    @Override
    public List<BookMeta> getContentList() {
        return bookList;
    }
}

File: book-model/src/main/java/org/gdgcd/demo/service/BookMetaEnvelope.kt
package org.gdgcd.demo.service


import com.fasterxml.jackson.annotation.JsonProperty

import java.util.ArrayList

class BookMetaEnvelope : Page<BookMeta> {
    @JsonProperty("Error")
    var errorNo: String = ""

    @JsonProperty("Time")
    var responseTime: Float? = null

    @JsonProperty("Total")
    var totalCount: String = ""

    @JsonProperty("Page")
    var pageNo: Int? = null

    @JsonProperty("Books")
    var bookList: List<BookMeta> = ArrayList()

    override fun getContentList(): List<BookMeta> {
        return bookList
    }
}
